tests/unit/useOnboardingStatus.test.tsx

```
@@ -0,0 +1,150 @@
+
+// We need a more dynamic mock to handle different return values per test
+let mockDbState: Record<string, number> = {};
+
+vi.mock('@/integrations/supabase/client', () => ({
+  supabase: {
+    from: vi.fn((table) => {
+        const count = mockDbState[table] ?? 0;
+        return {
+            select: vi.fn().mockReturnThis(),
+            eq: vi.fn().mockReturnThis(),
+            then: (resolve: any) => resolve({ count, error: null })
```

Biome (2.1.2)

[error] Do not add then to an object.



(lint/suspicious/noThenProperty)